src/decode.rs

```diff
@@ -4392,17 +4392,14 @@ pub(crate) unsafe fn rav1d_decode_frame_init(
         || need_cdef_lpf_copy != f.lf.need_cdef_lpf_copy
         || f.sbh != f.lf.cdef_buf_sbh
     {
-        rav1d_free_aligned(f.lf.cdef_line_buf as *mut c_void);
         let mut alloc_sz: usize = 64;
         alloc_sz += (y_stride.unsigned_abs() * 4 * f.sbh as usize) << need_cdef_lpf_copy;
         alloc_sz += (uv_stride.unsigned_abs() * 8 * f.sbh as usize) << need_cdef_lpf_copy;
-        f.lf.cdef_line_buf = rav1d_alloc_aligned(alloc_sz, 32) as *mut u8;
-        let mut ptr = f.lf.cdef_line_buf;
-        if ptr.is_null() {
-            f.lf.cdef_buf_plane_sz[1] = 0;
-            f.lf.cdef_buf_plane_sz[0] = f.lf.cdef_buf_plane_sz[1];
-            return Err(ENOMEM);
-        }
+        // TODO: Fallbile allocation. We need to do the following on allocation
+        // failure:
+        // f.lf.cdef_buf_plane_sz = [0, 0];
+        f.lf.cdef_line_buf.resize(alloc_sz, 0);
```

Comment:
`Rav1dFrameContext_lf::{cdef_buf_plane_sz, cdef_buf_sbh, need_cdef_lpf_copy}` can all be removed.  They're only used in checking if we need to resize or not, but we can fold that check into `cdef_line_buf.len()`, which is checked by `.resize` already.

This would be the simpler initialization code (can't make it a suggestion since the lines are too far):

```rust
    let need_cdef_lpf_copy = (c.tc.len() > 1 && has_resize != 0) as c_int;
    let y_sz = (y_stride.unsigned_abs() * 4 * f.sbh as usize) << need_cdef_lpf_copy;
    let uv_sz = (uv_stride.unsigned_abs() * 8 * f.sbh as usize) << need_cdef_lpf_copy;
    // TODO: Fallbile allocation
    f.lf.cdef_line_buf.resize(64 + y_sz + uv_sz, 0);
    let mut ptr = f.lf.cdef_line_buf.as_mut_ptr();

    ptr = ptr.offset(32);
    if y_stride < 0 {
        f.lf.cdef_line[0][0] = ptr.offset(-(y_stride * (f.sbh as isize * 4 - 1))) as *mut DynPixel;
        f.lf.cdef_line[1][0] = ptr.offset(-(y_stride * (f.sbh as isize * 4 - 3))) as *mut DynPixel;
    } else {
        f.lf.cdef_line[0][0] = ptr.offset(y_stride * 0) as *mut DynPixel;
        f.lf.cdef_line[1][0] = ptr.offset(y_stride * 2) as *mut DynPixel;
    }
    ptr = ptr.offset(y_stride.abs() * f.sbh as isize * 4);
    if uv_stride < 0 {
        f.lf.cdef_line[0][1] = ptr.offset(-(uv_stride * (f.sbh as isize * 8 - 1))) as *mut DynPixel;
        f.lf.cdef_line[0][2] = ptr.offset(-(uv_stride * (f.sbh as isize * 8 - 3))) as *mut DynPixel;
        f.lf.cdef_line[1][1] = ptr.offset(-(uv_stride * (f.sbh as isize * 8 - 5))) as *mut DynPixel;
        f.lf.cdef_line[1][2] = ptr.offset(-(uv_stride * (f.sbh as isize * 8 - 7))) as *mut DynPixel;
    } else {
        f.lf.cdef_line[0][1] = ptr.offset(uv_stride * 0) as *mut DynPixel;
        f.lf.cdef_line[0][2] = ptr.offset(uv_stride * 2) as *mut DynPixel;
        f.lf.cdef_line[1][1] = ptr.offset(uv_stride * 4) as *mut DynPixel;
        f.lf.cdef_line[1][2] = ptr.offset(uv_stride * 6) as *mut DynPixel;
    }

    if need_cdef_lpf_copy != 0 {
        ptr = ptr.offset(uv_stride.abs() * f.sbh as isize * 8);
        if y_stride < 0 {
            f.lf.cdef_lpf_line[0] =
                ptr.offset(-(y_stride * (f.sbh as isize * 4 - 1))) as *mut DynPixel;
        } else {
            f.lf.cdef_lpf_line[0] = ptr as *mut DynPixel;
        }
        ptr = ptr.offset(y_stride.abs() * f.sbh as isize * 4);
        if uv_stride < 0 {
            f.lf.cdef_lpf_line[1] =
                ptr.offset(-(uv_stride * (f.sbh as isize * 4 - 1))) as *mut DynPixel;
            f.lf.cdef_lpf_line[2] =
                ptr.offset(-(uv_stride * (f.sbh as isize * 8 - 1))) as *mut DynPixel;
        } else {
            f.lf.cdef_lpf_line[1] = ptr as *mut DynPixel;
            f.lf.cdef_lpf_line[2] = ptr.offset(uv_stride * f.sbh as isize * 4) as *mut DynPixel;
        }
    }
```

Reply:
Good catch. Done!